libkbfs/folder_block_ops.go

```diff
@@ -2551,12 +2566,14 @@ func (fbo *folderBlockOps) SearchForNodes(ctx context.Context,
 		return nodeMap, nil
 	}
 
-	// Start with the root node
 	rootPtr := md.data.Dir.BlockPointer
 	var node Node
 	if cache == fbo.nodeCache {
```

Comment:
I don't love having these specific corner cases that effectively test for cache invalidation. It seems like it could be pretty brittle. Can we instead actively invalidate `fbo.nodeCache`, so we don't have to deal with these specific checks in cache accesses?